xkcd/pkg/index/index.go

```diff
@@ -0,0 +1,158 @@
+package index
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/vespian/go-exercises/xkcd/pkg/pbuff"
+	"github.com/vespian/go-exercises/xkcd/pkg/types"
+	"github.com/vespian/go-exercises/xkcd/pkg/web"
+)
+
+func serialize(a types.AllStories, t types.OndiskSerialization,
+) (
+	blob []byte,
+	err error,
+) {
+	switch t {
+	case types.JSON:
+		blob, err = json.Marshal(a)
+		if err != nil {
+			return nil, fmt.Errorf("Index JSON marshaling failed: %s", err)
```

Comment:
Again from the "Clean Code": The name of a variable, function or class should answer the big questions such as why it exists, what it does, how it is used. This variable serves more as documentation, makes it easier for maintainer to read and understand code.